### DISCUSSION TITLE
After updating the version, the speed becomes very slow

### DISCUSSION THREAD

**Discussion by loveyou181** [aclap-dev/video-downloadhelper#55] [Help and Q&A]
### Your question

600MB takes 1 hour and 42 minutes to download. There was no such problem before the updated version.

After updating the version, the speed becomes very slow

### Addon & CoApp version and details

windows 10 

⊖Video DownloadHelper 8.1.6.0
Video DownloadHelper
版本 8.1.6.0
瀏覽器地區: zh-TW
穩定版
編譯時間: 2024-01-18
編譯選項: browser=chrome, linuxlic=false, noyt=true
⊖平台
平台 Win x86-64
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
⊖夥伴軟體已安裝
夥伴軟體版本: VdhCoApp 2.0.17
夥伴軟體執行檔: F:\DownloadHelper CoApp\vdhcoapp.exe

**Comment by GrampaWildWilly** [comment]
Let's have an example URL or two.

**Reply by loveyou181** [reply]
https://gimy.ai/eps/242773-7-1.html

**Reply by GrampaWildWilly** [reply]
All I get when I go to that page is this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d3f92347-e8d5-49e9-92d4-3137b6b1f3af)

Doesn't look like I can replicate your results, which I very much wanted to.  Let me make sure what that URL is.  That is a web page on which you are able to see a video.  If you wanted to, you could click a play button & make the video play on that page.  Is that correct?

**Reply by GrampaWildWilly** [reply]
I did a Google search for gimy.ai & the search results looked like this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a644c060-c902-494c-bfbf-60458724d66f)

So the site is a repository of TV shows & movies.  But somehow, it is blocked.  Given the perennial state of Sino-American diplomacy, I think it will be impossible for me to look at anything on that site.

My only suggestion is that you try the latest beta of VDH.  You are using Chrome so I don't know what the latest version number is for Chrome since no version numbers appear on the page where you can get the betas:

https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions

I can tell you that the latest VDH beta for Firefox is 8.2.0.19a1 because that is what I'm using.  You are using the latest version of the CoApp so I would think it makes sense to use the latest versi0n of the browser extension as well, & that would be a beta version.

**Comment by paulrouget** [comment]
> After updating the version

The version of the addon or the version of the coapp or both?

**Reply by loveyou181** [reply]
version: 8.1.6.0
beta: false
build date: 2024-01-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: zh-TW
license: accepted
key (only first 16 fisrt characters): fcd0ccff61304bf8
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "F:\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "win7",
    "arch": "x86_64",
    "node": "10"
  },
  "home": "C:\\Users\\USER",
  "converterBinary": "F:\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by MaxWundersam** [comment]
Hi @all, I have another example, it is by the way with almost ecery video clip on pornhub the same: very, very slow download since the latest update. Some work normal, most do not. Example: https://de.pornhub.com/view_video.php?viewkey=ph5faef58e8622a
Also, with some downloaded videos, there is no video preview image produced for the file in the windows explorer. The video will play, but it is just he generic item of the player. I am using firefox.

**Reply by MaxWundersam** [reply]
As an addition: if you can click on a file which shows the MB size, all downloads seem to work. Trouble with speed and missing preview image is solely happening with the stream options (1920 x ... etc.)

**Reply by GrampaWildWilly** [reply]
I went to that Pornhub page & tried to get VDH to download the 1280x720 variant that it presented.  I got 18 seconds of what looked like an extract from the middle of the scene.  Then I tried to get it directly with ffmpeg, without using VDH.  The ffprobe showed a duration of N/A, which means something is non-standard about this.  In addition, there was timed_ID3 data in it, something that ffmpeg ignores.  Despite the anomalies, I told ffmpeg to download this.  What I got was 46 seconds, once again from somewhere in the middle of this thing.  I got a lot of errors saying 403 Forbidden.  So I have to conclude that this particular video has something wrong with it.

I have observed some anomalous behavior on Pornhub (the US site, yours is the German site) on occasion.  Some things will download & some won't.  I have even had the weird experience that something that refused to download today downloaded fine tomorrow.  I can't explain it.  This does seem to be more of a problem with the site than with VDH.  If the site presents things in a standard way, VDH will download it.  If the site is being a problem, you're going to have trouble downloading the content.

**And then I had a hunch.**

When I am testing anything that other users post, I always make 2 changes to my Firefox settings:
. I set media.autoplay.blocking_policy in about:config to 2.  Normally I run with this set to 1 to prevent auto-play of content.  The value 2 is a stricter do-not-auto-play order that I have found is necessary on YouTube.
. I disable DRM.  If I'm going to test something for somebody, I want to find out up front whether the content is protected by DRM, because if it is, it's a waste of time trying to download it.

So I decided to revert the auto-play setting to 1 & turn DRM back on.  I suspect it is the setting value of 2 vs 1 rather than the DRM setting that made a difference.  I did not get the error from Firefox that it couldn't play the content because DRM was disabled, so that's why I don't think the DRM setting made any difference.  But suddenly, VDH downloaded this video no problem.

To make sure this wasn't a fluke, I picked another video at random whose thumbnail appears lower down on that page.  It happened to be this one:

https://de.pornhub.com/view_video.php?viewkey=ph6342f3ad5c4de

VDH downloaded this one as well just fine.  This one had a 1920x1080 variant & that's the one I downloaded.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.19a1 beta, CoApp 2.0.17.  You are running the same CoApp but you are not running the same VDH.  I suggest you try the beta & see how that works for you.

For me, I'm going to have to remember to leave the auto-play setting at 1 for everything except YouTube.  If I leave it at 1 on YouTube, things start playing automatically anyway, which I don't want.  But other sites seem to respect the value 1 & not auto-play.  So I need to stop switching that on every site, only YouTube.

**Reply by GrampaWildWilly** [reply]
Oh yeah.  Where is the beta?  Scroll this page to the top.  Look for the link labelled Wiki.  Click that.  You should be able to take things from there.

**Reply by GrampaWildWilly** [reply]
You also mentioned speed.  Yes, that site is not very fast.  I think I was getting something on the order of 500,000 bytes per second download speed from your site.  This is no better than I used to get before I upgraded my DSL connection to a proper high-speed service.  However, the downloads I did completed in reasonable time, probably because the videos were short and they have been compressed so the bit rates are quite low.  That makes for smaller files.  If you really want to discuss download speed, you need to monitor your downloads in the Resource Monitor, Network tab.  Resource Monitor is built into Windows.  Has been at least since W7.  Scrounge around.  You'll find it.  Try the system search field in the Task Bar or the search apps field in the Start Menu.

**Reply by MaxWundersam** [reply]
First of all, thank you very much for your reply. It is amazing to read what you all did to sort out this problem. I am not sure if got you right on  what to do exactly with those settings. I can try the beta though... but you faced the same issues with your version in the beginning, so I am not to confident. Can you explain, were I can adjust that settings in Firefox exactly? 
By the way, it is since the last update this way, before it was just fine on Pornhub. I found that - at least in some cases - the stream download also does not work on other pornsites (not related to Pornhub from my knowledge). There must have been a change in VDH, because all this worked fine some weeks ago. 
Concerning the missing preview icon, this happens mostly on Xhamster, I can also confirm, that this was working fine before. The fun thing is, that the video will often play anyway, but it makes me crazy if I have no preview icon for the video to distinguish those from the real broken ones. Weird... I do know that this exactly started with the latest update, 100% sure. I am running Video DownloadHelper 8.1.4.0.

**Reply by GrampaWildWilly** [reply]
There are 2 settings that I mess with in Firefox.  You are using Chrome.  You should probably ignore what I've said because it probably doesn't apply to you on your browser.  For anybody else coming in here & using Firefox, here's a more detailed explanation.

There are 2 settings that I play with.  The first is the checkbox for turning DRM support in the browser on & off.  I generally uncheck this box when I am looking at something a fellow user posts.  If the item in question shows me that the content of the page requires DRM, then I stop right there.  There is no point in investigating any problem with downloading that content.  It is protected by DRM.  VDH has never, does not now, & will never download content protected by DRM.  I turn the setting off as a time saver.  Is it protected by DRM?  Yes?  Stop investigating.

The other setting is in about:config.  It's one of the thousands upon thousands of poorly documented or completely undocumented settings that are in Firefox.  I found out about it via web searches.  It seems that over the course of a number of releases of Firefox, and not recently, Mozilla changed the way autoplay blocking was implemented.  The setting I mentioned above requires a value of 1 in order to block autoplay of any A/V content when you visit a page.  This works on almost all web sites.  The glaring exception?  Of course.  YouTube.  YouTube ignores the setting of 1.  It's supposed to prevent autoplay but it doesn't on YouTube.  So I have to set that Firefox parameter to a value of 2 in order to properly prevent YouTube pages from autoplaying their content.  I hate autoplay.  I don't want something playing that I'm going to download anyway.  Recent changes to VDH have allowed most things to be downloaded without first making them play or changing their resolution in the YouTube player.  So I set that parameter value to 2 for YouTube.

But I have been in the habit of changing that parameter value to 2 for sites I've investigated for fellow users.  This never caused a problem until I did it on German Pornhub.  Why, I can't explain.  But that's what happened.  However, the value of 2 is really meant only for YouTube.  So I am no longer going to be changing that value to 2 before I investigate some unknown site mentioned by a fellow user.  But it's a Firefox setting.  I'm sure the other browsers have other ways of blocking autoplay.  The issue with Firefox is that there are settings in the GUI dialog of settings that allege to block autoplay.  But they don't always, at least, not on YouTube.  At least, that was the case a couple of years ago when I first encountered the issue & went researching it.  It may no longer be the problem it was.  I'm not looking for the answer again.  I'm still setting the value to 2 before I visit YouTube & autoplay is blocked.  If this behavior ever changes, I'll have to research it again.

**Reply by GrampaWildWilly** [reply]
When it comes to thumbnails for videos, I have found that the file manager has drastically changed between Windows 7 & Windows 11.  I was using W7 for over 10 years.  At the beginning of December, barely 2 months ago, I completed my build of a new computer & did an install from scratch of W11.  So my experience is not deep with W11.  On W7, the bottom of the file manager window would show the preview thumbnail of a video file.  I would show you a screenshot but like I say, I'm not running W7 any more.  On W11, things look like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/aadd7afb-367c-428a-8c2f-c3032fcd7f10)

There's no thumbnail in the bottom of the window.  But if I click the thing in the upper right corner, the thing I'm indicating with that arrow, it changes to look like this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4b008eaa-dc37-4614-a2b6-1d995a6ade0d)

You get this huge panel on the right with so much information you have to scroll it to see it all.  But you do get the thumbnail.  This was so much neater, more compact on W7.  But I guess Microsoft would say it's new & improved.  I say it's worse.

There is another consideration.  I have been using something called K-Lite Codec Pack for a few years, meaning I started using it back on W7.  It improves the way Windows handles these thumbnails.  It also provides file properties for more types of A/V files than what basic Windows does.  I'm still using it on W11.  KLCP can turn the thumbnail display on & off.  So you might consider looking into that.  One handy tool in KLCP is something called Mediainfo.  It gives most of the same information that ffprobe does, but it does it in a little GUI app window, so it's more convenient to use.

So if you're looking for better thumbnail handling, you might consider KLCP.  But all of this has nothing at all to do with VDH.

**Reply by MaxWundersam** [reply]
Wow and again thanks a lot, I will dig through all this on the weekend and let you know my findings afterwards. Thank you very much! :-)

**Comment by paulrouget** [comment]
@MaxWundersam @loveyou181 

Thank you for sharing the page URLs.

@loveyou181 I'm still working on that.

@MaxWundersam I think the issue in your case is a little different. Not a slow download, but the progress is not properly displayed. Can you confirm if you wait a bit, it jumps from ~1% to 100% immediately?

**Reply by MaxWundersam** [reply]
Actually, no. It loads incredibly slow... even short video length takes ages. I mostly quit these after latest 30 mins. Sometimes, very seldom, it speeds up and gets slow again. It almost seems as if the site where you download from, sets you on slowest possible speed. Example: https://www.sunporno.com/videos/1521498/best-of-nichtmehr17-3

**Reply by GrampaWildWilly** [reply]
I went to that page & VDH showed me 3 variants.  The third one was only 2.5M, nowhere near big enough for a video lasting 1:20:39.  So I downloaded the other 2, since there wasn't enough information on the VDH menu for me to determine which one might be the video.  This is what I was getting:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a99e8526-f2ec-416f-8b51-19907e62140d)

Yup.  That's pretty slow.  It had been somewhat faster, although only about 500,000 - 600,000 bytes per second.  That's total usage for the 2 downloads combined.  But it settled back to this & it seems to be the equilibrium state for this site.  I don't believe this is a VDH issue.  Some sites are just slow.  This appears to be a slow site.

I posted now because I wanted to give some feedback.  The downloads are continuing & VDH is telling me now that each one will take a bit more than 40 minutes.  I'm guessing it's going to be 2 copies of the same video.  If I don't post again, assume the downloads completed fine & the videos played fine in VLC.

This is on Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.1.0a1 beta, CoApp 2.0.17.  Note the new VDH beta.

**Comment by MaxWundersam** [comment]
Most problems of this kind occur on Pornhub

**Reply by GrampaWildWilly** [reply]
While those German pornos were downloading, I went to Pornhub & chose these 2 at random:

https://www.pornhub.com/view_video.php?viewkey=ph62ac1b241c6cd
https://www.pornhub.com/view_video.php?viewkey=6599fbb47b837

The first one gave me a 232M mp4 of duration 11:47, resolution 1920x1080 after downloading for 12 seconds.  The second one gave me a 290M mp4 of duration 12:39, resolution 1920x1080 after downloading for 3:30.  But the second one paused for at least 2 long stretches when ffmpeg (meaning our CoApp) wasn't even listed in the Resource Monitor.  This is something we've been seeing here & there, that a download will inexplicably just stop & then suddenly resume after a long delay.

So I have to say Pornhub is better than Sunporno when it comes to speed, but it's inconsistent because of the delays.  When it's not delaying, Pornhub will give 10 million bytes per second or thereabouts.  But the effective speed can be greatly decreased by these inexplicable delays.  Sunporno is still plugging along at its pitiful slow speed, but it's plugging along.  I admit I'm not camped on the Resource Monitor staring at it, but I don't think Sunporno has delayed like Pornhub did.  I think we just have to live with what these servers give us.  Some are fast, some are slow, some like to delay for long stretches.  C'est la vie, mon chum.

**Reply by MaxWundersam** [reply]
I am with you an d thank you for the testing. The only thing which makes me wonder, is the fact that before the update, all Pornhub movies downloaded at the same speed w/o any delay. Weired coincidence maybe, but hard to believe it has nothing to do with it. Anyway, thank you again very much. Meanwhile my VDH wanted to update and I confirmed, seems to be a little bit better now, buit still there are the slow ones and still I do not get preview picture icons for the files. I will try something, which came into my mind and will report after. Keep ypu posted!

**Reply by GrampaWildWilly** [reply]
There may be hope.  Take a look at #149.  That approach might turn up some new information here.  I'll leave that to you as an exercise.

**Reply by paulrouget** [reply]
For pornhub, both the download strategy hasn't changed between 8.1 and 8.2. But we've seen some throttling happening lately (but not happening here, so I can't reproduce). The way we select variants has changed though. So maybe a problem at that level? No idea at the moment.

**Reply by MaxWundersam** [reply]
I will try that... were can one find the old versions? Do you have a link for me?

**Reply by paulrouget** [reply]
I'll be investigating that speed problem.

https://github.com/aclap-dev/video-downloadhelper/wiki/Older-versions

**Reply by WallyGelatau** [reply]
I've been finding that the problem is pretty random on PornHub.  Sometimes a video will download fast, sometimes they are extremely slow or get stuck close to the start, up to about 6%.  But if I refresh the page and try again, sometimes they will download fine and sometimes not.  Sometimes it takes 3 or 4 attempts. Generally I give up if not successful at that point.

**Reply by WallyGelatau** [reply]
Rolled back to 8.1.4.0. and the issue has gone away. ☺

**Reply by paulrouget** [reply]
This version improves the reported speed: https://github.com/aclap-dev/video-downloadhelper/discussions/199

**Comment by paulrouget** [comment]
Does that improve the download speed issue? https://github.com/aclap-dev/video-downloadhelper/discussions/199

**Reply by GrampaWildWilly** [reply]
The speed thing was never about what VDH reported on the blue dot status menu.  I monitor speed in the Windows Resource Monitor.  That's where I observe the putrid speeds & the minutes-long stoppages.  These appear to be the sites in question throttling the downloads.  I will say, however, that in a few other threads on here, which you no doubt have read, that I have done parallel VDH/CoApp and standalone ffmpeg tests.  My ffmpeg tests use the CoApp's ffmpeg.  I will see considerably faster downloads invoking ffmpeg myself than what I get in VDH.  Sometimes.  Every site & every download seems to be its own unique case.

**Comment by paulrouget** [comment]
I'm struggling to reproduce this here.

Something I'd like to understand:

- this started with version 8.2. Version 8.1 introduced a download method called Side Download. Can you confirm that in 8.1 you were using Side Download?
- I have made a minor fix in version 8.2.1.4 which, possible, might have improved the situation. If you use Firefox, can you give it a try: https://github.com/aclap-dev/video-downloadhelper/discussions/232

**Reply by GrampaWildWilly** [reply]
I just installed 8.2.1.5a1 beta.  When I tried the download on February 17 (see upthread) I was getting about 256K bytes per second download speed, as you can see in the image I posted.  I tried again just now & I was getting about 1M.  I killed the download after about a minute.  But 1M is 4 times what I was getting before.  It's still not particularly fast.  The VDH status message was telling me it would take another 25 minutes or so to complete.  But it's better than it was.
